File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  before_subscribe :set_room, :set_usuario

  def subscribed
    stream_from @chat_room
    ActionCable.server.broadcast(@chat_room, {
      mensagem: "Usuario #{@usuario} entrou"
    })
  end

  def receive(data)
    data['usuario'] = @usuario
    ActionCable.server.broadcast(@chat_room, data)
  end

  def set_room
    p 'configure'
    @chat_room = params[:chat]
  end

  def set_usuario
    @usuario = params[:usuario]
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "channels/consumer"

const chat = new URLSearchParams(window.location.search).get("chat")
const usuario = new URLSearchParams(window.location.search).get("usuario")

consumer.subscriptions.create({ channel: "ChatChannel", chat: chat, usuario: usuario }, {
    connected() {
        this.input = document.getElementById("input")
        this.chat = document.getElementById("chat")
        const button = document.getElementById("button")
        button.addEventListener("click", this.enviar_mensagem.bind(this))
    },

    disconnected() {
        // Called when the subscription has been terminated by the server
    },

    received(dados) {
        console.log(dados)
        if (dados.usuario != null) {
            this.chat.innerHTML += `<p>${dados.usuario}: ${dados.mensagem}</p>`
        } else {
            this.chat.innerHTML += `<p>${dados.mensagem}</p>`
        }
    },

    enviar_mensagem() {
        this.perform("receive", { mensagem: input.value })
        input.value = ""
    }
});
